### PR TITLE
feat: Add maxUnavailable PDB support to helm chart

### LINKED ISSUE #4238: Helm: PDB maxUnavailable support
*Description*:
>Describe the desired behavior, what scenario it enables and how it
would be used.
I'd like to be able to configure maxUnavailable in the PodDisruptionbBudget

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by jukie**
**What type of PR is this?**

**What this PR does / why we need it**:
Adds ability to set `maxUnavailable` in the PodDisruptionBudget

**Which issue(s) this PR fixes**:

Fixes #4238

### CODE REVIEW

File: charts/gateway-helm/templates/envoy-gateway-deployment.yaml [hunk 1]
```diff
@@ -1,25 +1,3 @@
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: envoy-gateway
-  namespace: '{{ .Release.Namespace }}'
-  labels:
-  {{- include "eg.labels" . | nindent 4 }}
----
-{{- if and .Values.podDisruptionBudget.minAvailable (ge (int .Values.podDisruptionBudget.minAvailable) 1) }}
```
Comment (arkodg) [inline charts/gateway-helm/templates/envoy-gateway-deployment.yaml:9]: prefer keeping this as this to make this backwards compatible and only inserting
```
maxUnavailable: {{ .Values.podDisruptionBudget.maxUnavailable }}
```

along with the `if` that will needed to ensure this template kicks in
cc @alexwo @guydc

Comment (jukie) [reply charts/gateway-helm/templates/envoy-gateway-deployment.yaml:9]: Do you mean you'd prefer having it always being applied, and if not input provided would default to minAvailable=0?

Comment (arkodg) [reply charts/gateway-helm/templates/envoy-gateway-deployment.yaml:9]: the top level `if` could we tweaked to `{{- if and .Values.podDisruptionBudget }}` i.e. create the resource, if any field is set

**Comment by arkodg** [comment]
can you also add a test or modify this one https://github.com/envoyproxy/gateway/blob/main/test/helm/gateway-helm/control-plane-with-pdb.in.yaml

File: charts/gateway-helm/templates/envoy-gateway-deployment.yaml (hunk 1 above)
Comment (jukie) [reply charts/gateway-helm/templates/envoy-gateway-deployment.yaml:9]: Makes sense! I can adjust it a bit more if preferred

**Comment by jukie** [comment]
@arkodg anything else I should add?

**Review by zirain — APPROVED** [review]

**Review by shawnh2 — APPROVED** [review]
LGTM

**Pull request merged**